In [45]:
from sklearn import linear_model
from sklearn.metrics import mean_absolute_error
import os
import numpy as np
import xgboost

/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)


In [10]:
train = np.load("./Inverter_1_2017-03-30/train.npy")
test = np.load("./Inverter_1_2017-03-30/test.npy")

In [11]:
train.shape , test.shape

((5144, 6), (1287, 6))

In [15]:
x_train , y_train = train[:,1:] , train[:,0].reshape((-1,1))
x_test , y_test = test[:,1:] , test[:,0].reshape((-1,1))
x_train.shape, y_train.shape, x_test.shape, y_test.shape

((5144, 5), (5144, 1), (1287, 5), (1287, 1))

In [20]:
lin_mod = linear_model.LinearRegression()
lin_mod.fit(x_train,y_train)
y_pred = lin_mod.predict(x_test)
mean_absolute_error(y_test,y_pred)

0.6550246157973469

In [44]:
lin_mod = linear_model.Lasso(alpha=0.01)
lin_mod.fit(x_train,y_train)
y_pred = lin_mod.predict(x_test)
mean_absolute_error(y_test,y_pred)

0.6425370385806931

In [47]:
lin_mod = xgboost.XGBRegressor()
lin_mod.fit(x_train,y_train)
y_pred = lin_mod.predict(x_test)
mean_absolute_error(y_test,y_pred)

0.596540187399982

| Algorithm | Test Set MAE|
| ------------- | ------------- |
|Bonsai | 0.6228
|Linear Regression | 0.655
|Lasso Regression | 0.6425
|XG Boost | 0.5965